### DISCUSSION TITLE
How to switch Views? While avoiding new instances

### DISCUSSION THREAD

**Discussion by MegaMech** [AvaloniaUI/Avalonia#12250] [Q&A] [answered]
After switching the view, I can't get a reference to the View to switch the view back to the original one.
Because in the xaml I have <views:MyOriginalView>

My current solution requires creating a new MenuView instance but I want to re-use the same instance:

MainWindow,axaml: `<Window><views:MenuView></Window>`
Behind.cs:
```csharp
public void OnGotoMenu()
{
    // If I could get the View from MenuViewModel and do Content = _menuView that would be prefereable.
      MenuVIew = menuView = new MenuView();
     DataContext = _viewModel.MenuViewModel;
     Content = menuView;
}
// If I do this, I just get a black screen with the class name printed out.
{
    DataContext = _viewModel.MenuViewModel;
    Content = _viewModel.MenuViewModel;
}
```

https://github.com/MegaMech/ScoreView/tree/master

**Comment by MegaMech** [comment, marked as answer]
Solved this using:
```csharp
public MenuView _menuView;

MainWindow() {
    _menuView = (MenuView)Content;
}

OnMyGoBackEvent() {

   Content = _menuView;
}

**Comment by timunie** [comment]
tbh I think it's better to use DataTemplates or a ViewLocator if possible, see: 

- https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-datatemplate-samples 
- https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/Routing/BasicViewLocatorSample
